mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndex.java

```diff
@@ -199,4 +205,7 @@ private void logIfNoMessageId(SearchResult searchResult) {
         }
     }
 
+    private RoutingKey toRoutingKey(MailboxId mailboxId) {
```

Comment:
what about having a list of Routing policies that we would reference?
We could have a RoutingPolicy.RouteByMailboxId, etc
That would make the code much easier to understand, don't you think?
We could also inject the decision info the constructor so that the indexing code don't have this responsibility (SRP power).